### PR TITLE
Add simulateDepositEth and simulateDepositERC20

### PR DESCRIPTION
**Pull request by zencephalon**

### CODE REVIEW

File: src/actions/public/simulateDepositETH.test.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { expect, test } from 'vitest'
+import { publicClientMainnet } from '../../_test/utils'
+import { base } from '@roninjin10/rollup-chains'
+import { simulateDepositETH } from './simulateDepositETH'
+
+test('default', async () => {
+  expect(
+    await simulateDepositETH(publicClientMainnet, {
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositETH.test.ts:8]: @zencephalon do we want to test that the return value here can be passed to writeDepositETH?

File: src/actions/public/simulateDepositETH.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  Chain,
+  Hex,
+  SimulateContractParameters,
+  SimulateContractReturnType,
+} from 'viem'
+import { l1StandardBridgeABI } from '@eth-optimism/contracts-ts'
+import { OpChainL2 } from '@roninjin10/rollup-chains'
+import { simulateContract } from 'viem/actions'
+
+type DepositETHParameters = {
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositETH.ts:13]: can we share this across share and write?

File: src/actions/public/simulateDepositETH.test.ts (hunk 1 above)
Comment (zencephalon) [reply src/actions/public/simulateDepositETH.test.ts:8]: yeah, I'm attempting to in a stashed commit

File: src/actions/public/simulateDepositETH.ts (hunk 2 above)
Comment (zencephalon) [reply src/actions/public/simulateDepositETH.ts:13]: Good idea

File: src/actions/public/simulateDepositETH.test.ts
Comment (wilsoncusack) [inline src/actions/public/simulateDepositETH.test.ts:1]: remember to add these to the public client decorator

File: src/actions/public/simulateDepositERC20.test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+} from 'viem/actions'
+
+const CBETHL1 = '0xbe9895146f7af43049ca1c1ae358b0541ea49704'
+const CBETHl2 = '0x2Ae3F1Ec7F1F5012CFEab0185bfc7aa3cf0DEc22'
+
+const USDCL1 = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'
+const USDCL2 = '0x2e668bb88287675e34c8df82686dfd0b7f0c0383'
+const zenaddress = '0xFd4F24676eD4588928213F37B126B53c07186F45'
+
+const approveABI = [
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositERC20.test.ts:20]: maybe just use erc20abi exported from wagmi

File: src/actions/public/simulateDepositERC20.test.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+import { accounts } from '../../_test/constants'
+import { simulateDepositERC20 } from './simulateDepositERC20'
+import {
+  writeContract,
+  getBalance,
+  simulateContract,
+  readContract,
+} from 'viem/actions'
+
+const CBETHL1 = '0xbe9895146f7af43049ca1c1ae358b0541ea49704'
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositERC20.test.ts:13]: think these are unused

File: src/actions/public/simulateDepositERC20.test.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+test('default', async () => {
+  await testClient.impersonateAccount({
+    address: zenaddress,
+  })
+  await testClient.setBalance({
+    address: zenaddress,
+    value: 1000000000000000000n,
+  })
+  const { request: approvalRequest } = await simulateContract(testClient, {
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositERC20.test.ts:72]: can just call writeContract here and not simulate 🤷

File: src/actions/public/simulateDepositERC20.test.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      l2Token: USDCL2,
+      amount: 1n,
+      gasLimit: 100000n,
+      data: '0x',
+    },
+    toChain: base,
+    account: zenaddress,
+  })
+
+  expect(request).toBeDefined()
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositERC20.test.ts:94]: do we want to test anything more? 🤔

File: src/types/depositERC20Parameters.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { Address, Hex } from 'viem'
+
+export type DepositERC20Parameters = {
```
Comment (wilsoncusack) [inline src/types/depositERC20Parameters.ts:3]: I am open to these each living in their own file, but I wonder it's best 🤔

File: src/actions/wallet/writeDepositETH.ts [hunk 8]
```diff
@@ -23,7 +18,7 @@ export type WriteDepositETHParameters<
 > = Omit<
   WriteContractParameters<
     typeof l1StandardBridgeABI,
-    'writeDepositETH',
```
Comment (wilsoncusack) [inline src/actions/wallet/writeDepositETH.ts:26]: this should probably have been breaking a test? Maybe write a test for it if not?

File: src/actions/public/simulateDepositETH.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    TChain,
+    TChainOverride
+  >,
+  'abi' | 'functionName' | 'args' | 'address'
+> & {
+  toChain: OpChainL2
+  args: DepositETHParameters
+}
+
+export type SimulateDepositETHReturnType<
```
Comment (wilsoncusack) [inline src/actions/public/simulateDepositETH.ts:29]: Do we need this 🤔 ? Can we not just use `SimulateContractReturnType`

Comment (zencephalon) [reply src/actions/public/simulateDepositETH.ts:29]: It makes the typing in publicOpStack ugly to not have this

File: src/actions/wallet/writeDepositETH.ts (hunk 8 above)
Comment (zencephalon) [reply src/actions/wallet/writeDepositETH.ts:26]: I'd expect it to have raised a type error in the tests we already had :/ not sure how to write a test for it, as it's a type level error

File: src/types/depositERC20Parameters.ts (hunk 7 above)
Comment (zencephalon) [reply src/types/depositERC20Parameters.ts:3]: I think if it gets out of hand we could merge them down the road

**Pull request merged**